### DISCUSSION TITLE
Text-decoration

### DISCUSSION THREAD

**Discussion by kmarler18** [tailwindlabs/tailwindcss#10144] [Help] [answered]
Hey All,

Hope you're well.

The various text-decoration properties work how I want them to in the Chrome browser on my laptop.

![Screen Shot 2022-12-21 at 5 37 39 PM](https://user-images.githubusercontent.com/96091818/209016989-8100fdbc-f4ab-412e-b45b-d0a3281d7fce.png)

And not how I want them to in Safari on iOS (note the underline color isn't yellow and isn't thick like it is above).

![209017095-a0db3a4a-2b26-4887-aa8b-d07126eb9356](https://user-images.githubusercontent.com/96091818/209364659-fd1e290a-bbe1-43eb-99f7-736634ae7c47.png)

Here's the code for the list items.

`<li class="mb-8 hover:text-red-1 hover:underline hover:underline-offset-4 hover:decoration-8 hover:decoration-yellow-1"><a href="#">...</a></li>`

My iPhone is running version 15.7.1 which as far as I understand means it's also running version 15 of Safari, which supports the decoration thickness and decoration color properties so I'm not sure why it shows up differently.

Any thoughts are appreciated.

Much love,
Kyle

**Comment by jayantghadge** [comment]
Try adding `-webkit `  prefix before the CSS property.
Check out the link below for more understanding -
[Browser Support - Tailwind CSS](https://tailwindcss.com/docs/browser-support)

**Comment by kmarler18** [comment, marked as answer]
@jayantghadge 

Thanks for your input!

Adding `-webkit-`didn't work, but I figured it out (sort of)...

Instead of putting the class names in the `li` element, I put them in the `a` element. 

Before: 
`<li class="mb-8 hover:text-red-1 hover:underline hover:underline-offset-4 hover:decoration-8 hover:decoration-yellow-1"><a href="#">...</a></li>`

After:
`<li><a class="inline-block mb-8 hover:text-red-primary hover:underline hover:underline-offset-4 hover:decoration-8 hover:decoration-yellow-primary" href="#">...</a></li>`

I still don't know _why_ it doesn't work when the class names are in `li`, but at least it works now. 

Any thoughts on the why?

Thanks a million.
Kyle
